pkg/cloudprovider/providers/vsphere/nodemanager.go

```diff
@@ -259,42 +261,57 @@ func (nm *NodeManager) GetNodeInfo(nodeName k8stypes.NodeName) (NodeInfo, error)
 		return nodeInfo
 	}
 	nodeInfo := getNodeInfo(nodeName)
+	var err error
 	if nodeInfo == nil {
-		err := nm.RediscoverNode(nodeName)
+		// Rediscover node if no NodeInfo found.
+		glog.V(4).Infof("No VM found for node %q. Initiating rediscovery.", convertToString(nodeName))
+		err = nm.RediscoverNode(nodeName)
 		if err != nil {
-			glog.V(4).Infof("error %q node info for node %q not found", err, convertToString(nodeName))
+			glog.Errorf("Error %q node info for node %q not found", err, convertToString(nodeName))
 			return NodeInfo{}, err
 		}
 		nodeInfo = getNodeInfo(nodeName)
+	} else {
+		// Renew the found NodeInfo to avoid stale vSphere connection.
+		glog.V(4).Infof("Renewing NodeInfo %q for node %q", nodeInfo, convertToString(nodeName))
```

Comment:
For printing `nodeInfo` we should use `%+v`

Reply:
Done.